lib/avrora/encoder.ex

```diff
@@ -100,7 +100,8 @@ defmodule Avrora.Encoder do
             48, 48, 48, 48, 45, 48, 48, 48, 48, 45, 48, 48, 48, 48, 48, 48, 48, 48, 48,
             48, 48, 48, 123, 20, 174, 71, 225, 250, 47, 64>>}
   """
-  @spec encode(map(), keyword(String.t())) :: {:ok, binary()} | {:error, term()}
+  @spec encode(map(), [{:schema_name, String.t()} | {:format, :guess | :registry | :ocf | :plain}]) ::
```

Comment:
Can I ask you to try this form?

```suggestion
  @spec encode(map(), schema_name: String.t(), format: :guess | :registry | :ocf | :plain) ::
```